databroker/src/grpc/kuksa_val_v2/val.rs

```diff
@@ -482,18 +625,18 @@ impl proto::val_server::Val for broker::DataBroker {
                                 match request {
                                     Some(req) => {
                                         match req.action {
-                                            Some(ProvidedActuation(_provided_actuation)) => {
-                                                if let Err(err) = response_stream_sender.send(Err(tonic::Status::new(tonic::Code::Unimplemented, "Unimplemented"))).await {
-                                                    debug!("Failed to send error response: {}", err);
-                                                }
-                                                break;
-                                            },
-                                            Some(PublishValuesRequest(publish_values_request)) => {
+                                           Some(PublishValuesRequest(publish_values_request)) => {
                                                 let response = publish_values(&broker, &publish_values_request).await;
                                                 if let Err(err) = response_stream_sender.send(Ok(response)).await
                                                 {
                                                     debug!("Failed to send response: {}", err);
                                                 }
+                                            },
+                                             Some(ProvideActuation(provided_actuation)) => {
+                                                if let Err(err) = provide_actuation(&broker, &provided_actuation, response_stream_sender).await {
+                                                    debug!("Failed to provide actuation: {}", err)
+                                                }
+                                                break;
                                             },
                                             Some(BatchActuateStreamResponse(_batch_actuate_stream_response)) => {
                                                 if let Err(err) = response_stream_sender.send(Err(tonic::Status::new(tonic::Code::Unimplemented, "Unimplemented"))).await {
```

Comment:
Handling here is unclear. What is expected to happen when a client sends the DataBroker a BatchActuateStreamResponse?